### ISSUE TITLE
rest_framework.filters.OrderingFilter doesn't pass context to serializers

### ISSUE THREAD

**Issue by mammique** [tomchristie/django-rest-framework#4541]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

Add 'rest_framework.filters.OrderingFilter' to 'DEFAULT_FILTER_BACKENDS' in settings.

Create a filter that calls the context:

```python
class MySerializer(serializers.HyperlinkedModelSerializer):

  def get_fields(self, *args, **kwargs):

        fields  = super(MySerializer, self).get_fields(*args, **kwargs)
        request = self.context['request']

        return fields
```

This will raise a KeyError on 'request'.

## Patch
I've fixed it, I'll send a pull request ASAP.

**Comment by tomchristie** [comment]
Seems pretty reasonable - thanks for this!
Are you able to come up with a different example that doesn't rely on overriding a bit of private API? (might help better demonstrate where & when this becomes an issue)

**Comment by mammique** [comment]
Hi Tom, thanks for your reply. No I just experienced this case, I have a serializer mixin similar to the one presented (overriding `get_fields()`) that uses request.user to filter fields for security, I use it everywhere in my code. As serializers are able to access context it worked like a charm everywhere in the API, until I enabled `filters.OrderingFilter` which calls `get_fields()` without providing the context, which breaks the browsable API (it still works in JSON because I doesn't need to test fields to generate the ordering widget). Sounds like a deficiency of `filters.OrderingFilter` to me, but maybe it is not…

Cheers,

Camille.
